src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/PostTaskSearchControllerCFTTest.java

```diff
@@ -448,6 +448,35 @@ public void should_return_a_200_with_search_results_based_on_state_unassigned()
         common.cleanUpTask(taskId);
     }
 
+    @Test
+    public void should_return_a_200_with_search_results_based_on_state_assigned() {
+        TestVariables taskVariables = common.setupTaskAndRetrieveIds();
+        String taskId = taskVariables.getTaskId();
+
+        common.setupOrganisationalRoleAssignment(authenticationHeaders);
+
+        insertTaskInCftTaskDb(taskVariables.getCaseId(), taskId);
+
+        SearchTaskRequest searchTaskRequest = new SearchTaskRequest(singletonList(
+            new SearchParameter(STATE, SearchOperator.IN, singletonList("assigned"))
+        ));
+
+        Response result = restApiActions.post(
+            ENDPOINT_BEING_TESTED + "?first_result=0&max_results=10",
+            searchTaskRequest,
+            authenticationHeaders
+        );
+
+        result.then().assertThat()
+            .statusCode(HttpStatus.OK.value())
+            .body("tasks.size()", lessThanOrEqualTo(10))
+            .body("tasks.jurisdiction", everyItem(is("IA")))
+            .body("tasks.task_state", everyItem(is("assigned")))
+            .body("total_records", greaterThanOrEqualTo(1));
```

Comment:
We might be able to also check the taskRoleResource has some expected values?

Reply:
not needed as discussed